### DISCUSSION TITLE
User login limit

### DISCUSSION THREAD

**Discussion by IceOmny** [Ylianst/MeshCentral#6250] [Q&A] [answered]
Hi all, first thanks for your time and support. To cut it short, I couldn't find the answer to this.

Is there a way to limit the login on some specific users or all user accounts? 
What I mean by the limit is that I don't want the same user account to be able to log in from multiple devices simultaneously. 
If possible, to logout the older user login, when there is a new successful login to the same account.

**Comment by si458** [comment, marked as answer]
`MaxSingleUserSessions: 1` add this under `limits` then add that under `domains`
![Screenshot_20240711_101348_Firefox](https://github.com/Ylianst/MeshCentral/assets/765314/26ec6303-d798-467b-9604-30856533fbc7)

**Reply by IceOmny** [reply]
I did, but it still can be logged on 2 devices at the same time. It doesn't log out the older season. Or did I do something wrong? It's under domains:

 "limits": {
        "type": "object",
        "properties": {
          "MaxDevices": {
            "type": "integer",
            "default": null,
            "description": "Maximum number of devices."
          },
          "MaxUserAccounts": {
            "type": "integer",
            "default": null,
            "description": "Maximum number of user accounts."
          },
          "MaxUserSessions": {
            "type": "integer",
            "default": null,
            "description": "Maximum number of user sessions."
          },
          "MaxAgentSessions": {
            "type": "integer",
            "default": null,
            "description": "Maximum number of agent sessions."
          },
          "MaxSingleUserSessions": {
            "type": "integer",
            "default": 1,
            "description": "Maximum number of sessions a single user can have. Each time a user opens a new browser tab or opens a new browser on a different computer, a new user session is created."
          }
        }
      }
    }
  },

**Reply by si458** [reply]
the screenshot is from the schema file which has all the valid options in including what they should be type wise, and default values and descriptions!

~~so you do something like this:~~
<img width="238" alt="image" src="https://github.com/user-attachments/assets/801f165c-83c2-4955-866c-13301f91417c">

DONT USE THAT USE THIS INSTEAD:
<img width="249" alt="image" src="https://github.com/user-attachments/assets/9a57d678-1a9f-4ade-8bc2-4ac8909681b4">

**Reply by IceOmny** [reply]
Thx, it works now. :)

Btw. is there an option to log out of earlier sessions when attempting to log in from a new device?

**Reply by si458** [reply]
> Thx, it works now. :)
> 
> Btw. is there an option to log out of earlier sessions when attempting to log in from a new device? 

No sadly, you would need to create a feature request for this and we can look at adding it in :)
